Description/Description.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/*
+ * This file is part of the Symfony CMF package.
+ *
+ * (c) 2011-2015 Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Component\Resource\Description;
+
+use Puli\Repository\Api\Resource\PuliResource;
+
+/**
+ * Descriptive metadata for resources.
+ */
+class Description
+{
+    /**
+     * @var array
+     */
+    private $descriptors = [];
+
+    /**
+     * @var PuliResource
+     */
+    private $resource;
+
+    /**
+     * @param PuliResource $resource
+     */
+    public function __construct(PuliResource $resource)
+    {
+        $this->resource = $resource;
+    }
+
+    /**
+     * Return the descriptors value for the given key.
+     *
+     * @param string $key
+     *
+     * @return mixed
```

Comment:
well, the consumer can use the result in any way they please - for example the result could be the name of an `.png` file to use as an icon, or it could be the URL, or a vendor-assigned human title for the resource class (e.g. "Page"). Its pretty generic.